avro-serializer/src/main/java/io/confluent/kafka/serializers/AvroData.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.serializers;
+
+import org.apache.avro.Conversions;
+import org.apache.avro.data.TimeConversions;
+import org.apache.avro.generic.GenericData;
+
+public class AvroData<T> {
+
+  public T getAvroDataWithConverters() {
```

Comment:
Shouldn't this method take a `GenericData` to be modified?

Reply:
And the method should be static, so perhaps `public static <T extends GenericData> T getAvroDataWithConverters(GenericData data)`

Reply:
I don't think a static method can be generalised this way. Anything since all avro data inherit one way or another from `GenericData`, I did what you suggested first, renamed the function, take a Generic as input parameter and made it static.